bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -796,7 +796,7 @@ impl Client {
             // because the UTD hook failed to load its data.
         }
 
-        *self.utd_hook.write().unwrap() = Some(Arc::new(utd_hook));
+        self.utd_hook.get_or_init(|| Arc::new(utd_hook));
```

Comment:
We'd probably want to throw an error if the utd hook manager was already set, because it could be super confusing for callers if they assumed that the UTD hook was overwritten, as before.

Let's add some note about that in this method's doc comment too, please.